spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/wavefront/WavefrontMetricsExportAutoConfiguration.java

```diff
@@ -68,4 +75,17 @@ public WavefrontMeterRegistry wavefrontMeterRegistry(WavefrontConfig wavefrontCo
 		return WavefrontMeterRegistry.builder(wavefrontConfig).clock(clock).wavefrontSender(wavefrontSender).build();
 	}
 
+	@Bean
+	@ConditionalOnBean(ApplicationTags.class)
+	@ConditionalOnClass(MeterRegistryCustomizer.class)
+	MeterRegistryCustomizer<WavefrontMeterRegistry> applicationTagsCustomizer(ApplicationTags applicationTags) {
```

Comment:
Forgive my Spring ignorance, but will this allow for multiple `MeterRegistryCustomizer<WavefrontMeterRegistry>` beans in the event a user wants to provide customizations in addition to this one?

Reply:
I believe multiple Customizers are allowed. Here's where they're streamed and `.customize()` is invoked: https://github.com/spring-projects/spring-boot/blob/e0b67889a8f70228227e06732f9973a1e9bc1a86/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MeterRegistryConfigurer.java#L74-L75

However, since my code has an `@Bean` annotation, I figured that means there can only be a singleton of `MeterRegistryCustomizer<WavefrontMeterRegistry>`.

Reply:
Multiple customisers are allowed. The singleton scope applies to each specific bean, not to each type of bean. In other words, your `applicationTagsCustomizer` will be a singleton but there could be any number of other beans that implement `MeterRegistryCustomizer<WavefrontMeterRegistry>` and there will be a single instance of each of those as well.

Reply:
@wilkinsona thanks so much for correcting my understanding!